[Docs] Added Instructions for building the documentations (#967)
bowtie-json-schema/bowtie: docs/contributing.rst

Feel free to file issues or pull requests as well if you wish something was better documented, as this too can help prioritize.

> Building the Documentation Locally

Reviewer: I think this seems like it can go combined in the previous section rather than with its own heading maybe.

As in:

```
Documentation is written in `Sphinx-flavored reStructuredText <sphinx:index>`, so you'll want to at least have a quick look at the `tutorial <sphinx:tutorial>` if you have never written documentation using it.

The code for the documentation is in the ``docs/`` directory of our :org:`github repository <bowtie>`.

etc.
```

Perhaps test out combining them. Will leave another comment or two about the specific content.

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> The code for the documentation is in the ``docs/`` directory of our :org:`github repository <bowtie>`.

Reviewer: There's already a `gh` role for the repository itself, so you can do

```suggestion
The code for the documentation is in the :gh:`docs directory of the Bowtie repository <tree/main/docs>`.
```

You can build the documentation locally by running:

.. code:: sh

    $ nox -s "docs(dirhtml)" -- dirhtml

> This will generate ``/dirhtml`` directory containing the documentation in html.

Reviewer: ```suggestion
which will generate a directory called ``dirhtml`` containing the built HTML.
```

Feel free to file issues or pull requests as well if you wish something was better documented, as this too can help prioritize.

> **Building the Documentation Locally**

Reviewer: If you want another layer of heading (which seems OK), you should use a heading rather than just bolding the text.

The syntax for that is:

```suggestion
Building the Documentation Locally
===========================
```

but I'll apply that change manually and merge.

Thanks again!
```
